src/border_agent/border_agent.hpp

```diff
@@ -140,25 +140,23 @@ class BorderAgent : private NonCopyable
         uint32_t ToUint32(void) const;
     };
 
-    void        Start(void);
-    void        Stop(void);
-    static void HandleMdnsState(void *aContext, Mdns::Publisher::State aState);
-    void        HandleMdnsState(Mdns::Publisher::State aState);
-    void        PublishMeshCopService(void);
-    void        UpdateMeshCopService(void);
-    void        UnpublishMeshCopService(void);
+    void Start(void);
+    void Stop(void);
+    void HandleMdnsState(Mdns::Publisher::State aState);
+    void PublishMeshCopService(void);
+    void UpdateMeshCopService(void);
+    void UnpublishMeshCopService(void);
 #if OTBR_ENABLE_DBUS_SERVER
     void HandleUpdateVendorMeshCoPTxtEntries(std::map<std::string, std::vector<uint8_t>> aUpdate);
 #endif
 
     void HandleThreadStateChanged(otChangedFlags aFlags);
 
-    bool IsThreadStarted(void) const;
-    bool IsPskcInitialized(void) const;
+    bool        IsThreadStarted(void) const;
+    std::string getAlternativeServiceInstanceName() const;
```

Comment:
If we want to make it a common utility, we need to specify current name as parameters:
```c++
std::string GetAlternativeServiceInstanceName(const std::string &aCurrentName) const;
```
And it will not be straightforward for us to create a new name based on current name, for example, we need to consider if current name is ready renamed.

That said, it will be much easier to accept a `aOriginalName` parameter name but that doesn't make sense and the user need to save the original name for this purpose.

This method can be simple if we keep it private to `BorderAgent` and we can think about making it public accessible when there is another usage. Thoughts?